### PR TITLE
Adding tests to IfSymbol Tuple

### PR DESCRIPTION
**Pull request by Chemaclass**
# Description

Adding tests to IfSymbol Tuple.

# Changes

- I refactored a little bit and I created some unit tests for the `Tuple\IfSymbol`

### CODE REVIEW

File: tests/php/Unit/Analyzer/TupleSymbol/IfSymbolTest.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        yield 'No arguments provided: (if)' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_IF)
+            ),
+        ];
+
+        yield 'Only one argument provided: (if one)' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_IF),
+                Symbol::create('one'),
```
Comment (Chemaclass) [inline tests/php/Unit/Analyzer/TupleSymbol/IfSymbolTest.php:42]: @jenshaase I got a question for you here: is it really necessary to decorate these literal values as Symbols? I mean, for example, the test will still pass(green) with these:
```php
'tuple' => Tuple::create(
    Symbol::NAME_IF,
    'one',
),
// OR
'tuple' => Tuple::create(
    Symbol::NAME_IF,
    'one',
    'two',
    'three',
    'four',
),
```

File: src/php/Analyzer/TupleSymbol/IfSymbol.php [hunk 2]
```diff
@@ -11,34 +11,51 @@
...
+    {
+        $envWithDisallowRecurFrame = $env
+            ->withContext(NodeEnvironment::CONTEXT_EXPRESSION)
+            ->withDisallowRecurFrame();
 
-        return new IfNode(
-            $env,
-            $testExpr,
-            $thenExpr,
-            $elseExpr,
```
Comment (Chemaclass) [inline src/php/Analyzer/TupleSymbol/IfSymbol.php:36]: I saw that we use in the [control flow docu](https://github.com/jenshaase/phel-lang/tree/master/doc/content/documentation/control-flow.md) this wording: test, then, else.
Screenshot 2020-10-24 at 15 01 55

But I think `(if condition truthy falsy)` is more expressive. What do you think? If you like the idea, I would create a PR renaming from the docu and related code 
- test -> condition
- then -> truthy
- else -> falsy

Comment (jenshaase) [reply src/php/Analyzer/TupleSymbol/IfSymbol.php:36]: In other Lisp documentations you will find the wording `test`, `then` and `else`. Therefore, I used it like they did.

File: tests/php/Unit/Analyzer/TupleSymbol/IfSymbolTest.php (hunk 1 above)
Comment (jenshaase) [reply tests/php/Unit/Analyzer/TupleSymbol/IfSymbolTest.php:42]: Yes you can do this. It doesn't matter for the test. But it is different Phel code and you may want to update it in the string above.

```
(if one two three four)
```
vs
```
(if "one" "two" "three" "four")
```

**Review by jenshaase — CHANGES_REQUESTED** [review]
Looks good. But leave the wording `test`, `then` and `else`.

File: src/php/Analyzer/TupleSymbol/IfSymbol.php (hunk 2 above)
Comment (Chemaclass) [reply src/php/Analyzer/TupleSymbol/IfSymbol.php:36]: Ah, then I will modify it in my PR as well to use this same naming

**Review by jenshaase — APPROVED** [review]
Cool! Thanks.

**Pull request merged**